plugin/go.vim

```diff
@@ -260,6 +260,16 @@ augroup vim-go
   " in the same window doesn't highlight the most recently matched
   " identifier's positions.
   autocmd BufWinEnter *.go call go#guru#ClearSameIds()
-augroup END
+
+  autocmd BufEnter *.go
```

Comment:
I saw some cases where `BufEnter` gets fired twice, but I was unable to duplicate it when I tried to.

In any event, it caused me to realize that if vim fires `BufEnter` twice, the second event will cause `b:old_gopath` to be overwritten with the current `$GOPATH`, and then when the buffer is left, `BufLeave` cannot assign the original value of `GOPATH`.  Can we add protection for this case by only checking `g:go_autodetect_gopath` when `if !exists('b:old_gopath')`